### ISSUE TITLE
Two scheduled events, fired at the same time, cause an error

### ISSUE THREAD

**Issue by jaklimoff** [adamcharnock/lightbus#20]
### Description
When there are two scheduled events, fired at the same time, error happens

### Example
```
import logging

import lightbus
import schedule as schedule
from lightbus.utilities.config import random_name

logger = logging.getLogger("lightbus")

bus = lightbus.create(
    service_name='scheduler',
    features=['tasks'],
)


class SomeApi(lightbus.Api):
    comments_check_requested = lightbus.Event(parameters=("contest",))
    likes_check_requested = lightbus.Event(parameters=("contest",))

    class Meta:
        name = "some"


bus.client.register_api(SomeApi())


@bus.client.schedule(schedule.every(3).seconds)
def collect_comments():
    logger.info("Comments!")
    bus.some.comments_check_requested.fire(contest=random_name(4))


@bus.client.schedule(schedule.every(3).seconds)
def collect_likes():
    logger.info("Likes!")
    bus.some.likes_check_requested.fire(contest=random_name(4))

```

### Logs

```
/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/bin/python /Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/bin/lightbus run -b src.workers.fail.bus
 lightbus.creation              | Importing bus.py from src.workers.fail.bus
 lightbus.creation              | No config file specified, will use default config
 lightbus.client                | Lightbus is setting up: service_name (set with -s or LIGHTBUS_SERVICE_NAME):scheduler, process_name (with with -p or LIGHTBUS_PROCESS_NAME):vizh
 lightbus.client                | Default transports are setup as follows:: RPC transport:RedisRpcTransport @ redis://127.0.0.1:6379/0, Result transport:RedisResultTransport @ redis://127.0.0.1:6379/0, Event transport:RedisEventTransport @ redis://127.0.0.1:6379/0, Schema transport:RedisSchemaTransport @ redis://127.0.0.1:6379/0
 lightbus.client                | No plugins loaded
 lightbus.client                | Enabled features (3): rpcs, events, tasks
 lightbus.client                | Disabled features (0): 
 lightbus.client                | APIs in registry (3): some, internal.state, internal.metrics
 lightbus.client                | Loaded the following remote schemas (4): some, internal.metrics, internal.state, contest
 lightbus.client                | Loaded the following local schemas (4): some, internal.state, internal.metrics
 lightbus.client                | Executing before_worker_start & on_start hooks...
 lightbus.client                | Execution of before_worker_start & on_start hooks was successful
 lightbus                       | Comments!
 lightbus                       | Likes!
 lightbus.client                | 📤  Sending event some.comments_check_requested
 lightbus.utilities.async_tools | ERROR: 'NoneType' object is not iterable (async_tools:128)
Traceback (most recent call last):
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/utilities/async_tools.py", line 121, in exception_handling_context
    yield
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/utilities/async_tools.py", line 108, in check_for_exception
    fut.result()
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/utilities/async_tools.py", line 196, in call_on_schedule
    await run_user_provided_callable(callback, args=[], kwargs={}, bus_client=bus_client)
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/utilities/async_tools.py", line 161, in run_user_provided_callable
    return await future
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/utilities/async_tools.py", line 159, in <lambda>
    executor=None, func=lambda: callable_(*args, **kwargs)
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/src/workers/fail/bus.py", line 35, in collect_likes
    bus.some.likes_check_requested.fire(contest=random_name(4))
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/path.py", line 119, in fire
    timeout=self.client.config.api(self.api_name).event_fire_timeout,
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/utilities/async_tools.py", line 47, in block
    raise e
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/utilities/async_tools.py", line 44, in block
    val = loop.run_until_complete(asyncio.wait_for(coroutine, timeout=timeout, loop=loop))
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
    return future.result()
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
    return fut.result()
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/path.py", line 131, in fire_async
    api_name=self.api_name, name=self.name, kwargs=kwargs, options=bus_options
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/client_worker.py", line 70, in wrapper
    new_traceback = traceback.format_list(new_stack)
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/traceback.py", line 39, in format_list
    return StackSummary.from_list(extracted_list).format()
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/traceback.py", line 377, in from_list
    for frame in a_list:
TypeError: 'NoneType' object is not iterable
Traceback (most recent call last):
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/bin/lightbus", line 10, in <module>
    sys.exit(lightbus_entry_point())
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/commands/__init__.py", line 32, in lightbus_entry_point
    run_command_from_args()
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/commands/__init__.py", line 62, in run_command_from_args
    parsed_args.func(parsed_args, config, plugin_registry, **extra)
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/commands/run.py", line 60, in handle
    self._handle(args, config, plugin_registry)
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/commands/run.py", line 115, in _handle
    bus.client.run_forever()
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/client.py", line 249, in run_forever
    self.stop_server()
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/client_worker.py", line 166, in wrapper
    return fn(*args, **kwargs)
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/client.py", line 356, in stop_server
    block(self._stop_server_inner())
  File "/Users/jaklimoff/Projects/trbr/incontest/incontest_api/.venv/lib/python3.7/site-packages/lightbus/client_worker.py", line 70, in wrapper
    new_traceback = traceback.format_list(new_stack)
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/traceback.py", line 39, in format_list
    return StackSummary.from_list(extracted_list).format()
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/traceback.py", line 377, in from_list
    for frame in a_list:
TypeError: 'NoneType' object is not iterable

Process finished with exit code 1
```

**Comment by adamcharnock** [comment]
Thank you for this Jack. Just looking it over quickly now - can you tell me
what lightbus verison you’re using? 0.10alpha has some major refactoring
which may resolve this.

**Comment by jaklimoff** [comment]
Oh indeed! Didn't notice it. I used version 0.9. Since 0.10 no errors! 

Thank you :)

P.S. Lightbus is an amazing tool!

**Comment by adamcharnock** [comment]
Thank you @jaklimoff! I'm very glad to hear you're finding it useful 😄 

It's a rainy day here so I may well get 0.10 stable pushed out today, and faff around with the docs a bit too.

**Comment by adamcharnock** [comment]
FYI @jaklimoff, I just released v0.10, which is exactly the same code as the 0.10alpha7 you are using.

Thanks for filing this issue, its great you took the time and I'm glad I already had a fix lined up. Please let me know if anything else comes up. I've been running this in production for a while, but there is every chance you'll find something I missed. You can get me here or on Discord.
